### PR TITLE
c7n kubernetes provider

### PR DESCRIPTION
**Pull request by kapilt**
work in progress from hackathon, still needs some testing infrastructure.

### CODE REVIEW

**Comment by ewbankkit** [comment]
The official Python client library for Kubernetes - https://github.com/kubernetes-client/python - which this work uses seems to have extensive unit tests. We should investigate what can be used.

File: tools/c7n_kube/c7n_kube/resources/namespace.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+from kubernetes.client import V1DeleteOptions
+
+
+@resources.register('namespace')
+class Namespace(QueryResourceManager):
+    class resource_type(TypeInfo):
+        group = 'Core'
+        version = 'V1'
+        enum_spec = ('list_namespace', 'items', {'watch': False})
```
Comment (kapilt) [inline tools/c7n_kube/c7n_kube/resources/namespace.py]: isn't `watch: False` a default?

File: tools/c7n_kube/setup.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    author_email="kapil.foss@gmail.com",
+    license="Apache-2.0",
+    packages=find_packages(),
+    entry_points={
+        "custodian.resources": [
+            'kube = c7n_kube.entry:initialize_kube']
+    },
+    install_requires=[
+        "c7n",
+        "kubernetes==6.0.0"
```
Comment (kapilt) [inline tools/c7n_kube/setup.py]: probably want to update to 7.0.0

**Comment by kapilt** [comment]
@ewbankkit i think the operative word wrt to the client tests was 'seems' to.. in practice it looks like smoke and mirrors as the test files are generated and the contents empty afaics for every test file 
 in https://github.com/kubernetes-client/python/blob/master/kubernetes/test/

**Comment by kapilt** [comment]
we can probably leverage some of the azure testing infrastructure here, both are to a common urllib3 adapter pattern, ie. reuse vcr here. not sure how it will play with the watch style method (websocket), but otherwise that should cover off on most api usage.

File: tools/c7n_kube/c7n_kube/provider.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+from c7n.provider import Provider, clouds
+
+from .client import Session
+from functools import partial
+
+import logging
+
+log = logging.getLogger('custodian.kube')
+
+@clouds.register('kube')
```
Comment (kapilt) [inline tools/c7n_kube/c7n_kube/provider.py]: thoughts on `kube` vs `k8s` for the provider name, if we're going to use an abbreviation, i figure let's use the one common in that community.

File: tools/c7n_kube/c7n_kube/resources/namespace.py (hunk 1 above)
Comment (thisisshi) [reply tools/c7n_kube/c7n_kube/resources/namespace.py]: yeah, watch shouldn't be necessary here

**Comment by kapilt** [comment]
hmm, looks like kubernetes client api is not compatible with python 3.7 due to use of async keyword.

**Comment by kapilt** [comment]
renamed provider to k8s, after some discussion with hackathon team that contributed this, going to go ahead and merge as is. intent will be going forward backfilling on test infrastructure (reusing azure vcr/requests integration).

**Pull request merged**